scan-bugs

```diff
@@ -0,0 +1,214 @@
+
+import contextlib
+import datetime
+import tempfile
+import os
+import shutil
+import subprocess
+import sys
+import HTMLParser
+
+master_repo = 'cockpit-project/cockpit'
+bug_status_output = "%s/bug_status.html" % os.environ['HOME']
+
+class IDParser(HTMLParser.HTMLParser):
+    def __init__(self, id):
+        self.id = id
+        self.result = None
+        self.in_tag = False
+        self.depth = {}
+        self.html = None
+        self.watch_startpos = False
+        HTMLParser.HTMLParser.__init__(self)
+
+    def loads(self, html):
+        self.html = html
+        self.feed(html)
+        self.close()
+
+    def handle_starttag(self, tag, attrs):
+        attrs = dict(attrs)
+        if self.in_tag:
+            self.find_startpos(None)
+        if 'id' in attrs and attrs['id'] == self.id:
+            self.result = [tag]
+            self.in_tag = True
+            self.watch_startpos = True
+        if self.in_tag:
+            if not tag in self.depth: self.depth[tag] = 0
+            self.depth[tag] += 1
+
+    def handle_endtag(self, tag):
+        if self.in_tag:
+            if tag in self.depth: self.depth[tag] -= 1
+            if self.depth[self.result[0]] == 0:
+                self.in_tag = False
+                self.result.append(self.getpos())
+
+    def find_startpos(self, x):
+        if self.watch_startpos:
+            self.watch_startpos = False
+            self.result.append(self.getpos())
+    handle_entityref = handle_charref = handle_data = handle_comment = handle_decl = handle_pi = unknown_decl = find_startpos
+
+    def get_result(self):
+        if self.result == None:
+            return None
+        if len(self.result) != 3:
+            return None
+        lines = self.html.split('\n')
+        lines = lines[self.result[1][0]-1:self.result[2][0]]
+        lines[0] = lines[0][self.result[1][1]:]
+        if len(lines) == 1:
+            lines[-1] = lines[-1][:self.result[2][1]-self.result[1][1]]
+        lines[-1] = lines[-1][:self.result[2][1]]
+        return '\n'.join(lines).strip()
+
+def get_element_by_id(id, html):
+    """ return tag reference by id in html """
+    parser = IDParser(id)
+    parser.loads(html)
+    return parser.get_result()
+
+
+@contextlib.contextmanager
+def tempdir(prefix='tmp'):
+    """A context manager for creating and then deleting a temporary directory."""
+    tmpdir = tempfile.mkdtemp(prefix=prefix)
+    try:
+        yield tmpdir
+    finally:
+        shutil.rmtree(tmpdir)
+
+
+def cmd(*args):
+    print "+", " ".join(list(args))
+    subprocess.check_call(list(args), stdout=sys.stdout, stderr=sys.stderr)
+
+def git(*args):
+    cmd("git", *args)
+
+def bug_details(bug_entry):
+    bug = { 'location': bug_entry[0],
+            'url': bug_entry[1] }
+    print "looking up bug %s" % (bug['url'])
+    html = subprocess.check_output ([ "curl", "-s", bug['url'] ])
+    bug['status'] = get_element_by_id("static_bug_status", html)
```

Comment:
Let's use "Unknown" and the url as defaults for status and description, respectively.  This makes the table a bit nicer for non-bugzilla URLs, like bugs.jquery.com.